diff -Naur a/rust/Makefile.am b/rust/Makefile.am
--- a/rust/Makefile.am	2023-10-06 04:31:57.605595900 +0700
+++ b/rust/Makefile.am	2023-10-07 09:39:49.696142900 +0700
@@ -81,6 +81,7 @@
 
 if HAVE_CBINDGEN
 gen/rust-bindings.h: $(RUST_SURICATA_LIB)
+	PATH=$(CARGO_HOME)/bin:$(PATH) \
 	cbindgen --config $(abs_top_srcdir)/rust/cbindgen.toml \
 		--quiet --verify --output $(abs_top_builddir)/rust/gen/rust-bindings.h || true
 else
@@ -92,6 +93,7 @@
 
 if HAVE_CBINDGEN
 dist/rust-bindings.h:
+	PATH=$(CARGO_HOME)/bin:$(PATH) \
 	cbindgen --config $(abs_top_srcdir)/rust/cbindgen.toml \
 		--quiet --output $(abs_top_builddir)/rust/dist/rust-bindings.h
 else
