Index: parodus-1.0.2/CMakeLists.txt
===================================================================
--- parodus-1.0.2.orig/CMakeLists.txt
+++ parodus-1.0.2/CMakeLists.txt
@@ -57,7 +57,7 @@ add_definitions("-DGIT_COMMIT_TAG=\"${GI
 add_definitions(-std=c99)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_GNU_SOURCE -DNOPOLL_LOGGER ")
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror -Wall -Wno-missing-field-initializers")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror -Wall -Wno-missing-field-initializers -Wno-error=address")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall")
 
 
@@ -73,7 +73,7 @@ ExternalProject_Add(trower-base64
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/trower-base64
     GIT_REPOSITORY https://github.com/Comcast/trower-base64.git
     GIT_TAG "v1.1.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
 )
 add_library(libtrower-base64 STATIC SHARED IMPORTED)
 add_dependencies(libtrower-base64 trower-base64)
@@ -85,9 +85,10 @@ ExternalProject_Add(nopoll
     PREFIX ${PREFIX_DIR}/nopoll
     GIT_REPOSITORY https://github.com/Comcast/nopoll.git
     GIT_TAG "1.0.1"
-    CONFIGURE_COMMAND COMMAND <SOURCE_DIR>/autogen.sh --prefix=${PREFIX}
+    CONFIGURE_COMMAND COMMAND CC=${CMAKE_C_COMPILER} <SOURCE_DIR>/autogen.sh --prefix=${PREFIX}
                                                      --includedir=${INCLUDE_DIR}
                                                      --libdir=${LIBRARY_DIR}
+                                                     --host=${CMAKE_SYSTEM_PROCESSOR}
                                                      ${CUSTOM_HOST}
     BUILD_IN_SOURCE 1
 )
@@ -101,7 +102,7 @@ ExternalProject_Add(nanomsg
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/nanomsg
     GIT_REPOSITORY https://github.com/nanomsg/nanomsg.git
     GIT_TAG "1.1.2"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
 )
 add_library(libnanomsg STATIC SHARED IMPORTED)
 add_dependencies(libnanomsg nanomsg)
@@ -114,7 +115,7 @@ ExternalProject_Add(cJSON
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/cJSON
     GIT_REPOSITORY https://github.com/DaveGamble/cJSON.git
     GIT_TAG "v1.7.8"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
 )
 add_library(libcJSON STATIC SHARED IMPORTED)
 add_dependencies(libcJSON cJSON)
@@ -126,7 +127,9 @@ ExternalProject_Add(msgpack
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/msgpack
     GIT_REPOSITORY https://github.com/msgpack/msgpack-c.git
     GIT_TAG "cpp-3.1.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} 
+                  -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} 
+                  -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
                   -DMSGPACK_ENABLE_CXX=OFF
                   -DMSGPACK_BUILD_EXAMPLES=OFF
 )
@@ -140,7 +143,10 @@ ExternalProject_Add(cimplog
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/cimplog
     GIT_REPOSITORY https://github.com/Comcast/cimplog.git
     GIT_TAG "1.0.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} 
+                  -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} 
+                  -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
+                  -DCMAKE_C_FLAGS=-Wno-error=format
 )
 add_library(libcimplog STATIC SHARED IMPORTED)
 add_dependencies(libcimplog cimplog)
@@ -153,7 +159,9 @@ ExternalProject_Add(wrp-c
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/wrp-c
     GIT_REPOSITORY https://github.com/Comcast/wrp-c.git
     GIT_TAG "1.0.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} 
+                  -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                  -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
                   -DMSGPACK_ENABLE_CXX=OFF
                   -DMSGPACK_BUILD_EXAMPLES=OFF
                   -DBUILD_TESTING=OFF
@@ -193,7 +201,14 @@ ExternalProject_Add(cjwt
     GIT_REPOSITORY https://github.com/Comcast/cjwt.git
     GIT_TAG "1b023c41bb2d6dbbf493c202ed81f06c84d5b51b"
     #GIT_TAG "1.0.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
+    PATCH_COMMAND sed -i "s+<cJSON.h>+<cjson/cJSON.h>+g" src/cjwt.c &&
+                  sed -i "s+<cJSON.h>+<cjson/cJSON.h>+g" src/cjwt.h &&
+                  sed -i "s+<base64.h>+<trower-base64/base64.h>+g" src/cjwt.c
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} 
+                  -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} 
+                  -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} 
+                  -DBUILD_TESTING=OFF
+                  -DCMAKE_C_FLAGS=-Wno-error=deprecated-declarations
 )
 add_library(libcjwt STATIC SHARED IMPORTED)
 add_dependencies(libcjwt cjwt)
Index: parodus-1.0.2/src/ParodusInternal.h
===================================================================
--- parodus-1.0.2.orig/src/ParodusInternal.h
+++ parodus-1.0.2/src/ParodusInternal.h
@@ -136,7 +136,6 @@ typedef struct {
 /*----------------------------------------------------------------------------*/
 extern bool g_shutdown;
 extern ParodusMsg *ParodusMsgQ;
-int numLoops;
 /*----------------------------------------------------------------------------*/
 /*                             Function Prototypes                            */
 /*----------------------------------------------------------------------------*/
Index: parodus-1.0.2/src/main.c
===================================================================
--- parodus-1.0.2.orig/src/main.c
+++ parodus-1.0.2/src/main.c
@@ -39,7 +39,8 @@
 /*----------------------------------------------------------------------------*/
 /*                            File Scoped Variables                           */
 /*----------------------------------------------------------------------------*/
-/* none */
+int numLoops;
+
 
 /*----------------------------------------------------------------------------*/
 /*                             Function Prototypes                            */
