Index: parodus-2.0.0/src/CMakeLists.txt
===================================================================
--- parodus-2.0.0.orig/src/CMakeLists.txt
+++ parodus-2.0.0/src/CMakeLists.txt
@@ -38,7 +38,7 @@ add_executable(parodus ${SOURCES})
 target_link_libraries (parodus
     ${CMAKE_THREAD_LIBS_INIT}
     -lwrp-c
-    -lmsgpackc
+    -lmsgpack-c
     -lcurl
     -ltrower-base64
     -lnopoll
Index: parodus-2.0.0/CMakeLists.txt
===================================================================
--- parodus-2.0.0.orig/CMakeLists.txt
+++ parodus-2.0.0/CMakeLists.txt
@@ -66,7 +66,7 @@ add_definitions("-DGIT_COMMIT_TAG=\"${GI
 add_definitions(-std=c99)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_GNU_SOURCE -DNOPOLL_LOGGER ")
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror -Wall -Wno-missing-field-initializers")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror -Wall -Wno-missing-field-initializers -Wno-error=address")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall")
 
 
@@ -74,117 +74,55 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}
 #-------------------------------------------------------------------------------
 find_package (Threads)
 
+# Find pkgconfig
+find_package(PkgConfig REQUIRED)
+
 if (NOT BUILD_YOCTO)
 
 # base64 external dependency
 #-------------------------------------------------------------------------------
-ExternalProject_Add(trower-base64
-    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/trower-base64
-    GIT_REPOSITORY https://github.com/Comcast/trower-base64.git
-    GIT_TAG "v1.1.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
-)
-add_library(libtrower-base64 STATIC SHARED IMPORTED)
-add_dependencies(libtrower-base64 trower-base64)
-
+find_library(LIBBASE64_LIBRARY NAMES libtrower-base64.so)
+if (NOT LIBBASE64_LIBRARY)
+    message(FATAL_ERROR "libtrower-base64 not found")
+endif (NOT LIBBASE64_LIBRARY)
 
 # nopoll external dependency
 #-------------------------------------------------------------------------------
-ExternalProject_Add(nopoll
-    PREFIX ${PREFIX_DIR}/nopoll
-    GIT_REPOSITORY https://github.com/Comcast/nopoll.git
-    GIT_TAG "1.0.3"
-    CONFIGURE_COMMAND COMMAND <SOURCE_DIR>/autogen.sh --prefix=${PREFIX}
-                                                     --includedir=${INCLUDE_DIR}
-                                                     --libdir=${LIBRARY_DIR}
-                                                     ${CUSTOM_HOST}
-    BUILD_IN_SOURCE 1
-)
-add_library(libnopoll STATIC SHARED IMPORTED)
-add_dependencies(libnopoll nopoll)
-
+find_library(LIBNOPOLL_LIBRARY NAMES libnopoll.so)
+if (NOT LIBNOPOLL_LIBRARY)
+    message(FATAL_ERROR "libnopoll not found")
+endif (NOT LIBNOPOLL_LIBRARY)
 
 # nanoMsg external dependency
 #-------------------------------------------------------------------------------
-ExternalProject_Add(nanomsg
-    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/nanomsg
-    GIT_REPOSITORY https://github.com/nanomsg/nanomsg.git
-    GIT_TAG "1.1.4"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
-)
-add_library(libnanomsg STATIC SHARED IMPORTED)
-add_dependencies(libnanomsg nanomsg)
-
+find_library(LIBNANOMSG_LIBRARY NAMES libnanomsg.so)
+if (NOT LIBNANOMSG_LIBRARY)
+    message(FATAL_ERROR "libnanomsg not found")
+endif (NOT LIBNANOMSG_LIBRARY)
 
 # cJSON external dependency
 #-------------------------------------------------------------------------------
-ExternalProject_Add(cJSON
-    #PREFIX ${PREFIX_DIR}/cJSON
-    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/cJSON
-    GIT_REPOSITORY https://github.com/DaveGamble/cJSON.git
-    GIT_TAG "v1.7.8"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
-)
-add_library(libcJSON STATIC SHARED IMPORTED)
-add_dependencies(libcJSON cJSON)
-
+pkg_check_modules(CJSON REQUIRED libcjson)
 
 # msgpack-c external dependency
 #-------------------------------------------------------------------------------
-ExternalProject_Add(msgpack
-    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/msgpack
-    GIT_REPOSITORY https://github.com/msgpack/msgpack-c.git
-    GIT_TAG "cpp-3.1.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
-                  -DMSGPACK_ENABLE_CXX=OFF
-                  -DMSGPACK_BUILD_EXAMPLES=OFF
-)
-add_library(libmsgpack STATIC SHARED IMPORTED)
-add_dependencies(libmsgpack msgpack)
-
+find_library(LIBMSGPACK_LIBRARY NAMES libmsgpackc.so)
 
 # curl external dependency
 #-------------------------------------------------------------------------------
-ExternalProject_Add(curl
-    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/curl
-    GIT_REPOSITORY https://github.com/curl/curl.git
-    GIT_TAG "curl-7_63_0"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
-)
-add_library(libcurl STATIC SHARED IMPORTED)
-add_dependencies(libcurl curl)
-
+find_package(CURL REQUIRED)
+include_directories(${CURL_INCLUDE_DIRS})
 
 # cimplog external dependency
 #-------------------------------------------------------------------------------
-ExternalProject_Add(cimplog
-    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/cimplog
-    GIT_REPOSITORY https://github.com/Comcast/cimplog.git
-    GIT_TAG "1.0.2"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
-)
-add_library(libcimplog STATIC SHARED IMPORTED)
-add_dependencies(libcimplog cimplog)
-
+find_library(LIBCIMPLOG_LIBRARY NAMES libcimplog.so)
 
 # wrp-c external dependency
 #-------------------------------------------------------------------------------
-ExternalProject_Add(wrp-c
-	DEPENDS trower-base64 msgpack cimplog
-    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/wrp-c
-    GIT_REPOSITORY https://github.com/xmidt-org/wrp-c.git
-    GIT_TAG "71f8a39fe39f98da007ed4cdabbb192be1da1685"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
-                  -DMSGPACK_ENABLE_CXX=OFF
-                  -DMSGPACK_BUILD_EXAMPLES=OFF
-                  -DBUILD_TESTING=OFF
-                  -DMAIN_PROJ_BUILD=ON
-                  -DMAIN_PROJ_LIB_PATH=${LIBRARY_DIR}
-    		  -DMAIN_PROJ_INCLUDE_PATH=${INCLUDE_DIR}
-)
-add_library(libwrp-c STATIC SHARED IMPORTED)
-add_dependencies(libwrp-c wrp-c)
-
+find_library(LIBWRP_C_LIBRARY NAMES libwrp-c.so)
+if (NOT LIBWRP_C_LIBRARY)
+    message(FATAL_ERROR "libwrp-c not found")
+endif (NOT LIBWRP_C_LIBRARY)
 
 if (ENABLE_SESHAT)
 # libseshat external dependency
@@ -209,15 +147,11 @@ endif (ENABLE_SESHAT)
 
 # libcjwt external dependency
 #-------------------------------------------------------------------------------
-ExternalProject_Add(cjwt
-    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/cjwt
-    GIT_REPOSITORY https://github.com/Comcast/cjwt.git
-    GIT_TAG "1b023c41bb2d6dbbf493c202ed81f06c84d5b51b"
-    #GIT_TAG "1.0.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
-)
-add_library(libcjwt STATIC SHARED IMPORTED)
-add_dependencies(libcjwt cjwt)
+find_library(LIBCJWT_LIBRARY NAMES libcjwt.so)
+
+if (NOT LIBCJWT_LIBRARY)
+    message(FATAL_ERROR "libcjwt not found")
+endif (NOT LIBCJWT_LIBRARY)
 
 if (FEATURE_DNS_QUERY)
 # libucresolv external dependency
Index: parodus-2.0.0/src/ParodusInternal.h
===================================================================
--- parodus-2.0.0.orig/src/ParodusInternal.h
+++ parodus-2.0.0/src/ParodusInternal.h
@@ -39,7 +39,7 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 
-#include <cJSON.h>
+#include <cjson/cJSON.h>
 #include <nopoll.h>
 #include <nanomsg/nn.h>
 #include <nanomsg/pipeline.h>
Index: parodus-2.0.0/src/crud_internal.c
===================================================================
--- parodus-2.0.0.orig/src/crud_internal.c
+++ parodus-2.0.0/src/crud_internal.c
@@ -19,7 +19,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <cJSON.h>
+#include <cjson/cJSON.h>
 #include <wrp-c.h>
 #include "crud_tasks.h"
 #include "crud_internal.h"
Index: parodus-2.0.0/src/crud_tasks.c
===================================================================
--- parodus-2.0.0.orig/src/crud_tasks.c
+++ parodus-2.0.0/src/crud_tasks.c
@@ -1,7 +1,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <cJSON.h>
+#include <cjson/cJSON.h>
 #include <wrp-c.h>
 #include "crud_tasks.h"
 #include "crud_internal.h"
