Index: parodus-85a0d9b7/CMakeLists.txt
===================================================================
--- parodus-85a0d9b7.orig/CMakeLists.txt
+++ parodus-85a0d9b7/CMakeLists.txt
@@ -65,7 +65,7 @@ add_definitions(-std=c99)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_GNU_SOURCE -DNOPOLL_LOGGER ")
 
 if (DEVICE_CAMERA)
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-error=all -Wno-missing-field-initializers")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-error=all -Wno-missing-field-initializers -Wimplicit-function-declaration")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=all")
 add_definitions(-DDEVICE_CAMERA)
 else ()
@@ -89,7 +89,7 @@ ExternalProject_Add(trower-base64
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/trower-base64
     GIT_REPOSITORY https://github.com/Comcast/trower-base64.git
     GIT_TAG "v1.1.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}  -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
 )
 add_library(libtrower-base64 STATIC SHARED IMPORTED)
 add_dependencies(libtrower-base64 trower-base64)
@@ -97,18 +97,18 @@ add_dependencies(libtrower-base64 trower
 
 # nopoll external dependency
 #-------------------------------------------------------------------------------
-ExternalProject_Add(nopoll
-    PREFIX ${PREFIX_DIR}/nopoll
-    GIT_REPOSITORY https://github.com/Comcast/nopoll.git
-    GIT_TAG "1.0.3"
-    CONFIGURE_COMMAND COMMAND <SOURCE_DIR>/autogen.sh --prefix=${PREFIX}
-                                                     --includedir=${INCLUDE_DIR}
-                                                     --libdir=${LIBRARY_DIR}
-                                                     ${CUSTOM_HOST}
-    BUILD_IN_SOURCE 1
-)
-add_library(libnopoll STATIC SHARED IMPORTED)
-add_dependencies(libnopoll nopoll)
+# ExternalProject_Add(nopoll
+#     PREFIX ${PREFIX_DIR}/nopoll
+#     GIT_REPOSITORY https://github.com/Comcast/nopoll.git
+#     GIT_TAG "1.0.3"
+#     CONFIGURE_COMMAND COMMAND <SOURCE_DIR>/autogen.sh --prefix=${PREFIX}
+#                                                      --includedir=${INCLUDE_DIR}
+#                                                      --libdir=${LIBRARY_DIR}
+#                                                      ${CUSTOM_HOST}
+#     BUILD_IN_SOURCE 1
+# )
+# add_library(libnopoll STATIC SHARED IMPORTED)
+# add_dependencies(libnopoll nopoll)
 
 
 # nanoMsg external dependency
@@ -117,7 +117,7 @@ ExternalProject_Add(nanomsg
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/nanomsg
     GIT_REPOSITORY https://github.com/nanomsg/nanomsg.git
     GIT_TAG "1.1.4"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
 )
 add_library(libnanomsg STATIC SHARED IMPORTED)
 add_dependencies(libnanomsg nanomsg)
@@ -130,7 +130,7 @@ ExternalProject_Add(cJSON
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/cJSON
     GIT_REPOSITORY https://github.com/DaveGamble/cJSON.git
     GIT_TAG "v1.7.8"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
 )
 add_library(libcJSON STATIC SHARED IMPORTED)
 add_dependencies(libcJSON cJSON)
@@ -142,9 +142,10 @@ ExternalProject_Add(msgpack
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/msgpack
     GIT_REPOSITORY https://github.com/msgpack/msgpack-c.git
     GIT_TAG "cpp-3.1.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
                   -DMSGPACK_ENABLE_CXX=OFF
                   -DMSGPACK_BUILD_EXAMPLES=OFF
+                  -DCMAKE_INSTALL_LIBDIR=${LIBRARY_DIR}
 )
 add_library(libmsgpack STATIC SHARED IMPORTED)
 add_dependencies(libmsgpack msgpack)
@@ -156,7 +157,10 @@ ExternalProject_Add(curl
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/curl
     GIT_REPOSITORY https://github.com/curl/curl.git
     GIT_TAG "curl-7_63_0"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} 
+                  -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} 
+                  -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} 
+                  -DBUILD_TESTING=OFF
 )
 add_library(libcurl STATIC SHARED IMPORTED)
 add_dependencies(libcurl curl)
@@ -168,7 +172,11 @@ ExternalProject_Add(cimplog
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/cimplog
     GIT_REPOSITORY https://github.com/Comcast/cimplog.git
     GIT_TAG "1.0.2"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} 
+                  -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} 
+                  -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} 
+                  -DBUILD_TESTING=OFF
+                  -DCMAKE_C_FLAGS=-Wno-error=format
 )
 add_library(libcimplog STATIC SHARED IMPORTED)
 add_dependencies(libcimplog cimplog)
@@ -181,7 +189,9 @@ ExternalProject_Add(wrp-c
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/wrp-c
     GIT_REPOSITORY https://github.com/xmidt-org/wrp-c.git
     GIT_TAG "b5ef4d10cb39905908788bc89ab3e4dab201db8a"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} 
+                  -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                  -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
                   -DMSGPACK_ENABLE_CXX=OFF
                   -DMSGPACK_BUILD_EXAMPLES=OFF
                   -DBUILD_TESTING=OFF
@@ -201,7 +211,8 @@ ExternalProject_Add(libseshat
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/libseshat
     GIT_REPOSITORY https://github.com/Comcast/seshat.git
     GIT_TAG "1.0.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
+    CMAKE_ARGS +=  -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                  -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
     	          -DMAIN_PROJ_BUILD=ON
     	          -DMAIN_PROJ_LIB_PATH=${LIBRARY_DIR}
     	          -DMAIN_PROJ_LIB64_PATH=${LIBRARY_DIR64}
@@ -219,9 +230,13 @@ endif (ENABLE_SESHAT)
 ExternalProject_Add(cjwt
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/_prefix/cjwt
     GIT_REPOSITORY https://github.com/Comcast/cjwt.git
-    GIT_TAG "1b023c41bb2d6dbbf493c202ed81f06c84d5b51b"
+    GIT_TAG "1f28bdaf9f79904d9459c6d5f537b7e234b872db"
     #GIT_TAG "1.0.1"
-    CMAKE_ARGS += -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} -DBUILD_TESTING=OFF
+    CMAKE_ARGS += -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} 
+                  -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                  -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR} 
+                  -DBUILD_TESTING=OFF
+                  -DCMAKE_C_FLAGS=-Wno-error=deprecated-declarations
 )
 add_library(libcjwt STATIC SHARED IMPORTED)
 add_dependencies(libcjwt cjwt)
Index: parodus-85a0d9b7/src/config.c
===================================================================
--- parodus-85a0d9b7.orig/src/config.c
+++ parodus-85a0d9b7/src/config.c
@@ -134,7 +134,7 @@ unsigned int get_algo_mask (const char *
 	{
 		algo_str = get_tok (algo_str, ':', tok, BUFLEN);
 		alg_val = cjwt_alg_str_to_enum (tok);
-		if ((alg_val < 0)  || (alg_val >= num_algorithms)) {
+		if ((alg_val < 0)  || (alg_val >= alg_last)) {
        ParodusError("Invalid jwt algorithm %s\n", tok);
        return (unsigned int) (-1);
 		}
Index: parodus-85a0d9b7/src/token.c
===================================================================
--- parodus-85a0d9b7.orig/src/token.c
+++ parodus-85a0d9b7/src/token.c
@@ -163,7 +163,7 @@ bool validate_algo(const cjwt_t *jwt)
         int alg = jwt->header.alg;
         int alg_mask;
 
-	if ((alg < 0) || (alg >= num_algorithms))
+	if ((alg < 0) || (alg >= alg_last))
 		return false;
 	alg_mask = 1<<alg;
 	if ((alg_mask & cfg->jwt_algo) == 0) {
Index: parodus-85a0d9b7/src/conn_interface.c
===================================================================
--- parodus-85a0d9b7.orig/src/conn_interface.c
+++ parodus-85a0d9b7/src/conn_interface.c
@@ -232,7 +232,6 @@ void createSocketConnection(void (* init
     nopoll_cleanup_library();
     curl_global_cleanup();
     clear_metadata();
-    rdk_logger_deinit();
     free_server_list(&server_list);
 }
 
