Index: rosenpass-0.2.1/Cargo.toml
===================================================================
--- rosenpass-0.2.1.orig/Cargo.toml	2023-11-18 22:16:22.000000000 +0000
+++ rosenpass-0.2.1/Cargo.toml	2024-02-15 13:20:10.617248814 +0000
@@ -8,3 +8,6 @@
 [workspace.metadata.release]
 # ensure that adding `--package` as argument to `cargo release` still creates version tags in the form of `vx.y.z`
 tag-prefix = ""
+
+[patch.crates-io]
+oqs-sys = { git = "https://github.com/bacnh85/liboqs-rust.git", branch = "v0.8.0_rosenpass" }
